jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/skaffold/SyncMapTask.java

```diff
@@ -0,0 +1,91 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle.skaffold;
+
+import com.google.cloud.tools.jib.filesystem.TempDirectoryProvider;
+import com.google.cloud.tools.jib.gradle.GradleProjectProperties;
+import com.google.cloud.tools.jib.gradle.GradleRawConfiguration;
+import com.google.cloud.tools.jib.gradle.JibExtension;
+import com.google.cloud.tools.jib.plugins.common.ContainerizingMode;
+import com.google.cloud.tools.jib.plugins.common.InvalidContainerizingModeException;
+import com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor;
+import com.google.common.base.Preconditions;
+import javax.annotation.Nullable;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.TaskAction;
+
+/**
+ * Prints out a map of local files to their location on the container.
+ *
+ * <p>Expected use: {@code ./gradlew _jibSkaffoldSyncMap -q} or {@code ./gradlew
+ * :<subproject>:_jibSkaffoldSyncMap -q}
+ */
+public class SyncMapTask extends DefaultTask {
+
+  @Nullable private JibExtension jibExtension;
+
+  public SyncMapTask setJibExtension(JibExtension jibExtension) {
+    this.jibExtension = jibExtension;
+    return this;
+  }
+
+  @TaskAction
+  public void listFilesAndTargets() {
+    Preconditions.checkNotNull(jibExtension);
+
+    try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider()) {
+      GradleProjectProperties projectProperties =
+          GradleProjectProperties.getForProject(getProject(), getLogger(), tempDirectoryProvider);
+
+      GradleRawConfiguration configuration = new GradleRawConfiguration(jibExtension);
+
+      // TODO: move these shared checks with SyncMapMojo into plugins-common
+      if (projectProperties.isWarProject()) {
+        throw new GradleException(
+            "Skaffold sync is currently only available for 'jar' style Jib projects, but the project "
+                + getProject().getName()
+                + " is configured to generate a 'war'");
+      }
+      try {
+        if (!ContainerizingMode.EXPLODED.equals(
+            ContainerizingMode.from(jibExtension.getContainerizingMode()))) {
+          throw new GradleException(
+              "Skaffold sync is currently only available for Jib projects in 'exploded' containerizing mode, but the containerizing mode of "
+                  + getProject().getName()
+                  + " is '"
+                  + jibExtension.getContainerizingMode()
+                  + "'");
+        }
+      } catch (InvalidContainerizingModeException ex) {
+        throw new GradleException("Invalid containerizing mode", ex);
+      }
+
+      try {
+        String syncMapJson =
+            PluginConfigurationProcessor.getSkaffoldSyncMap(configuration, projectProperties);
+
+        System.out.println();
```

Comment:
Makes sense. Perhaps we should apply this for all other tasks. I once thought Java is portable to the extend that it takes a special care of `\n` in strings to be protable (for example, I know from my experience, a string of `"/unix/style/path/"` just works as intended on Windows), but perhaps it doesn't. In any case, I think it's best to avoid using `\n` in strings.

I'll fix other tasks.